### PR TITLE
Can't enable SSL with MariaDB driver library. (#1182)

### PR DESCRIPTION
**Pull request by vakuum**
Please see #1182 for details.

### CODE REVIEW

**Comment by junaruga** [comment]
Thanks for the PR.

Could you add new CI case for mysql-8.0 server and mariadb client?

**Comment by junaruga** [comment]
@vakuum I tried to create the commit to create DB server: MySQL, DB client: MariaDB #1186 . You can pick up if you like.

I think the new CI case could create `mysql2.so` linking to the MariaDB client library. But I do not see new error. We might need to add a unit test to test this issue.

```
Run ldd lib/mysql2/mysql2.so
...
	libmariadb.so.3 => /usr/lib/x86_64-linux-gnu/libmariadb.so.3 (0x00007f73d1daf000)
...

**Comment by vakuum** [comment]
@junaruga I added your commit 0f44a2e44d64f85513a546626334ab33645f02b8 to the pull request.

**Comment by junaruga** [comment]
@vakuum Thanks for the work! I am sorry. Could you remove the 2nd commit? I am sorry. Because I got a feedback that it is not certain if we support the new case of the DB server (MySQL) + DB client (MariaDB) by adding the CI case. https://github.com/brianmario/mysql2/pull/1186#issuecomment-817189606 .

And right now our Travis CI has MariaDB (server and client) cases. But the macro `HAVE_CONST_MYSQL_OPT_SSL_ENFORCE` is not set on the cases. I think that's way your error case is not captured by our current CI cases. https://github.com/brianmario/mysql2/issues/1182#issuecomment-817303008

So, I think the right thing about the CI, is to add the case enabling `HAVE_CONST_MYSQL_OPT_SSL_ENFORCE` macro to a Travis CI mariadb case. But right now I have no idea about how to add it.

**Comment by junaruga** [comment]
Could you add my another commit to this PR? https://github.com/brianmario/mysql2/issues/1182#issuecomment-821072178 .

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
           - {os: ubuntu-16.04, ruby: 2.4, db: mariadb10.0, allow-failure: true}
           # Comment out due to ci/setup.sh stucking.
           # - {os: ubuntu-18.04, ruby: 2.4, db: mariadb10.1}
-          # `service mysql restart` fails.
+          # Allow failure due to the issue #965, #1165.
```
Comment (junaruga) [inline .github/workflows/build.yml:34]: I opened a PR #1193 to fix #965 and #1165 . After the PR will be merged, possibly the MariaDB 10.3 case passes.
And we can remove the comment line and `allow-failure: true` too.

**Comment by vakuum** [comment]
@junaruga: Is there any plan to merge this PR?

**Comment by junaruga** [comment]
@vakuum Sorry I do not have a permission to merge a PR on this repository.

**Comment by junaruga** [comment]
@vakuum if you want a new mysql2 gem including this PR now, I can recommend a way to create a releasing branch including this PR's commits on your forked repository, then run `gem build, mysql2.spec` to create the `mysql2-<version>.gem` to be used in a meantime.

**Comment by sodabrew** [comment]
Sorry for the delay to review, this is a great PR, thank you @vakuum !

**Pull request merged**